presto-spark-base/src/main/java/com/facebook/presto/spark/util/PrestoSparkStatsCollectionUtils.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.util;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.common.RuntimeMetric;
+import com.facebook.presto.common.RuntimeStats;
+import com.facebook.presto.common.RuntimeUnit;
+import com.facebook.presto.execution.TaskInfo;
+import com.facebook.presto.operator.NativeExecutionInfo;
+import com.facebook.presto.operator.OperatorStats;
+import com.facebook.presto.operator.PipelineStats;
+import com.facebook.presto.operator.TaskStats;
+import org.apache.commons.text.CaseUtils;
+import org.apache.spark.executor.TaskMetrics;
+import org.apache.spark.util.AccumulatorV2;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Set;
+
+public class PrestoSparkStatsCollectionUtils
+{
+    public static final String SPARK_INTERNAL_ACCUMULATOR_PREFIX = "internal.metrics.";
+    public static final String PRESTO_NATIVE_OPERATOR_STATS_SEP = "root.";
+    public static final String PRESTO_NATIVE_OPERATOR_STATS_PREFIX = "velox.";
+    private static final Logger log = Logger.get(PrestoSparkStatsCollectionUtils.class);
+
+    private PrestoSparkStatsCollectionUtils() {}
+
+    public static void collectMetrics(final TaskInfo taskInfo)
+    {
+        int taskId = -1;
+        int stageId = -1;
+        try {
+            taskId = taskInfo.getTaskId().getId();
+            stageId = taskInfo.getTaskId().getStageExecutionId().getStageId().getId();
+            Set<RuntimeStats> runtimeStatsSet = collectRuntimeStats(taskInfo);
+            collectMetrics(runtimeStatsSet);
+        }
+        catch (Exception e) {
+            log.error("An error occurred while processing taskId=%s  stageId=%s", taskId, stageId, e);
+        }
+    }
+
+    public static void collectMetrics(Set<RuntimeStats> runtimeStatsSet)
+    {
+        runtimeStatsSet.forEach(runStats ->
+        {
+            runStats.getMetrics().entrySet().forEach(entry -> {
+                String prestoKey = entry.getKey();
+                String sparkInternalAccumulatorKey = getSparkInternalAccumulatorKey(prestoKey);
```

Comment:
Yes, because velox add prefix. such as MetricSource. root to every key.

Reply:
As we discussed in the chat, could we change the accumulator names in Spark java side to match the exact name populated from the C++ side? I think it's error-prone manipulating/preprocessing the name in the middle of the workflow.

Reply:
This is different from the prefix we discussed. these prefixes are appended to the metrics by a different module.
Can we remove these prefixes without breaking other applications ? I will look into that.

Reply:
I believe these are the Velox metrics in fb-spark we discussed, if that's the case, as our discussion, currently no one is using it (the previous project which created these metrics at first place has already been deprecated) so we have all freedom to change the name on java side to match the name populated from C++.

Reply:
These are not Velox metrics.   

If cpp client stat key is set to KKK   runtimeStats key after deserialization will be  
shufflewrite.root.KKK  I am just removing  that prefix using root as separator. 

I did search in codeHub I couldn't find the source of the prefix.

Reply:
in your case, the `shufflewrite` is the velox plan node name and `root` is the velox plan node id for the `shuffleWrite` operator.  In general for all the metrics we got from velox, different records will be associated with different operator names (it could be "TableScan", "TableWrite", "shuffleWrite", "shuffleRead" etc) with different plan node ids (not just "root"),  so the current implementation in `getSparkInternalAccumulatorKey ` doesn't look right to me. I'd still suggest to just keeping what names we get from C++ as it is and changed the names on spark java side to match them.